src/main/java/technology/tabula/extractors/BatchSelectionExtractor.java

```diff
@@ -0,0 +1,473 @@
+						// original document
+						if (!textFound) {// && ocr){
+							System.out.println("Possible image based document: " + currentFile.getAbsolutePath());
+
+							// attempt to ocr document and processs again
+							if (ocrAllowed) {
+								System.out.println("Attempting to convert document to text based format...");
+
+								try {
+									OcrConverter ocr = new OcrConverter();
+
+									ocr.extract(currentFile.getAbsolutePath(), true);
+
+									File ocrFile = new File(currentFile.getAbsolutePath().substring(0,
+											currentFile.getAbsolutePath().length() - 4) + "_OCR.pdf");
+
+									deleteList.add(ocrFile.getAbsolutePath());
+
+									pdfDocument = PDDocument.load(ocrFile);
+
+									reading = true;
+
+									extractor = new ObjectExtractor(pdfDocument);
+
+									pageIterator = extractor.extract();
+
+									textFound = false;
+
+									textFound = scanDocForMatchedTables(basicExtractor, pageIterator, overlapThreshold, processType,
+											stringList, coordList, pageList, tables, tableHeaders);
+
+									pdfDocument.close();
+
+									reading = false;
+
+									// add document to list of created files instead? clean up later?
+									// can this trigger an exception too?
+									// need another try-catch?
+								} catch (Exception e) {
+									// e.printStackTrace();
+									System.out.println("Unable to properly convert or extract data from document");
+								}
+
+							} else
+								System.out.println("Ignored document based on input parameters");
+						}
+
+						// just in case
+						if (reading) {
+							pdfDocument.close();
+							reading = false;
+						}
+						// write data to new file
+						// currently overwrites existing files and creates a
+						// 	file even if no tables extracted
+						// may be better to check if file already exists
+						Writer writer = null;
+						BufferedWriter bufferedWriter = null;
+						File testFile = new File(
+								outputPath + "\\" + fileName.substring(0, fileName.lastIndexOf(".")) + ".csv");
+						testFile.createNewFile();
+						FileWriter fileWriter = new FileWriter(testFile);
+						bufferedWriter = new BufferedWriter(fileWriter);
+						// assuming CSV format for demo
+						writer = new CSVWriter();
+
+						int tableNum = 0; // needed?
+
+						// add header for OCR'd files?
+						for (Table table : tables) {
+							List<Table> fauxTableList = new ArrayList<Table>();
+							fauxTableList.add(table);
+
+							// write table header
+							bufferedWriter.write(tableHeaders.get(tableNum) + "\n");
+
+							// write single table
+							writer.write(bufferedWriter, fauxTableList);
+
+							// write line break
+							bufferedWriter.write("\n");
+
+							// increment table number?
+							tableNum++;
+						}
+
+						// shut the door on your way out
+						bufferedWriter.close();
+						// pdfDocument.close();
+					}
+
+					catch (Exception e) {
+						e.printStackTrace();
```

Comment:
If swallowing all possible Exceptions, please explain why. This is likely incorrect. Also, do not print stack trace to console.